app/src/main/java/com/example/taskyy/ui/AgendaScreen.kt

```diff
@@ -1,12 +1,203 @@
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.AbsoluteRoundedCornerShape
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material3.DatePicker
+import androidx.compose.material3.DatePickerDefaults
+import androidx.compose.material3.DisplayMode
+import androidx.compose.material3.DropdownMenu
+import androidx.compose.material3.DropdownMenuItem
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.Surface
 import androidx.compose.material3.Text
+import androidx.compose.material3.rememberDatePickerState
 import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.navigation.NavController
+import com.example.taskyy.R
+import com.example.taskyy.domain.navigation.Screen
+import com.example.taskyy.ui.events.AgendaEvent
+import com.example.taskyy.ui.viewmodels.AgendaState
+import java.time.LocalDateTime
+import java.time.ZoneId
 
 @Composable
-fun AgendaScreen(){
-    Column {
-        Text(text = "Agenda Screen")
+fun AgendaScreen(state: AgendaState, navController: NavController, onEvent: (AgendaEvent) -> Unit){
+    Column(
+        verticalArrangement = Arrangement.SpaceAround,
+        modifier = Modifier
+            .fillMaxSize()
+            .background(Color.Black)
+    ) {
+        Row (
+            horizontalArrangement = Arrangement.SpaceBetween,
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(horizontal = 10.dp)
+        ) {
+            Row(
+                horizontalArrangement = Arrangement.Start,
+                modifier = Modifier.clickable {
+                    onEvent(AgendaEvent.OnMonthExpanded(!state.isMonthExpanded))
+                }
+            ) {
+                MonthText(
+                    selectedMonth = state.selectedMonth
+                )
+                MonthDownArrow()
+                if (state.isMonthExpanded) {
+                    ShowDatePicker(
+                        onItemSelected = { selectedMonth ->
+                            onEvent(AgendaEvent.OnDateSelected(selectedMonth))
+                        },
+                    )
+                }
+            }
+            CircleWithInitials(
+                isUserDropDownExpanded = state.isUserDropDownExpanded,
+                onUserInitialsClicked = { isUserDropDownExpanded ->
+                    onEvent(AgendaEvent.OnUserInitialsClicked(isUserDropDownExpanded))
+                },
+                onItemSelected = {
+                    onEvent(AgendaEvent.OnLogOutCLicked)
+                },
+            )
+        }
```

Comment:
Your screen is quite big here which is why I'd recommend to split it up into more smaller composables. This piece here could be the `AgendaToolbar` for example. Separating a screen by such sections makes it much easier to scan its content